unity/PitayaExample/Assets/Pitaya/PitayaClient.cs

```diff
@@ -20,23 +20,14 @@ public class PitayaClient : IDisposable, IPitayaClient, IPitayaListener
         private TypeSubscriber<uint> _typeRequestSubscriber;
         private TypeSubscriber<string> _typePushSubscriber;
 
-        public PitayaClient()
-        {
-            Init(null, false, false, false, DEFAULT_CONNECTION_TIMEOUT);
-        }
-
-        public PitayaClient(int connectionTimeout)
-        {
-            Init(null, false, false, false, connectionTimeout);
-        }
-
-        public PitayaClient(string certificateName = null)
-        {
-            Init(certificateName, certificateName != null, false, false, DEFAULT_CONNECTION_TIMEOUT);
-        }
-
-        public PitayaClient(bool enableReconnect = false, string certificateName = null, int connectionTimeout = DEFAULT_CONNECTION_TIMEOUT)
-        {
+        public PitayaClient() : this(false) {}
+        public PitayaClient(int connectionTimeout) : this(false, null, connectionTimeout: connectionTimeout) {}
+        public PitayaClient(string certificateName = null) : this(false, certificateName: certificateName) {}
+        
+        public PitayaClient(bool enableReconnect = false, string certificateName = null, int connectionTimeout = DEFAULT_CONNECTION_TIMEOUT, IPitayaQueueDispatcher queueDispatcher = null)
+        {
+            if (queueDispatcher == null) queueDispatcher = MainQueueDispatcher.Create();
+            PitayaBinding.QueueDispatcher = queueDispatcher;
```

Comment:
There are some race conditions here if you're creating PitayaClients multiple times here, since the queue dispatcher could be used while you do the switch.

Reply:
I don't think it's different than before, `Create()` returns a single instance. Maybe I can rename it for better clarity.
```c#
        public static MainQueueDispatcher Create(bool dontDestroyOnLoad = true)
        {
            if (_instance != null)
                return _instance;
            
            var go = new GameObject (typeof(MainQueueDispatcher).FullName);
            _instance = go.AddComponent<MainQueueDispatcher> ();
            if (dontDestroyOnLoad) DontDestroyOnLoad(_instance.gameObject);
            return _instance;
        }
```

Reply:
Fair enough, I think we can leave it that way.